File: backend/src/main/java/com/gxs/bank/model/ContactSubmission.java
package com.gxs.bank.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "contact_submissions")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class ContactSubmission {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String email;

    private String subject;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String message;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (status == null) status = Status.NEW;
    }

    public enum Status {
        NEW, IN_PROGRESS, RESOLVED, CLOSED
    }
}

File: backend-python/com/gxs/bank/model/ContactSubmission.py
from datetime import datetime
from enum import Enum
import uuid

from sqlalchemy import Column, DateTime, Enum as SQLEnum, String, Text

from com.gxs.bank.config.database import Base


class Status(str, Enum):
    NEW = "NEW"
    IN_PROGRESS = "IN_PROGRESS"
    RESOLVED = "RESOLVED"
    CLOSED = "CLOSED"


class ContactSubmission(Base):
    __tablename__ = "contact_submissions"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    name = Column(String(255), nullable=False)
    email = Column(String(255), nullable=False)
    subject = Column(String(255), nullable=True)
    message = Column(Text, nullable=False)
    status = Column(SQLEnum(Status), nullable=False, default=Status.NEW)
    createdAt = Column("created_at", DateTime, nullable=False, default=datetime.utcnow)


ContactSubmission.Status = Status
